### PR TITLE
configurable gpt action

### PR DESCRIPTION
**Pull request by maheshsattala**
Changes Made:
1. Added KaironFaqAction and CRUD Operations with APIs 
2. Added methods in ActionUtility
3. Made Changes related to GPT3
4. Added/Modified unit and Integration test for the same

### CODE REVIEW

File: kairon/shared/data/processor.py [hunk 1]
```diff
@@ -4677,6 +4681,71 @@ def get_two_stage_fallback_action_config(self, bot: Text, name: Text = KAIRON_TW
             action.pop('user')
             yield action
 
+    def add_kairon_faq_action(self, request_data: dict, bot: Text, user: Text):
```
Comment (udit-pandey) [inline kairon/shared/data/processor.py:4684]: add validation that faq feature is enabled

File: tests/integration_test/services_test.py [hunk 2]
```diff
@@ -907,6 +908,157 @@ def test_get_content_not_exists():
...
+        headers={"Authorization": pytest.token_type + " " + pytest.access_token},
+    )
+    actual = response.json()
+    assert actual["success"]
+    assert actual["error_code"] == 0
+    assert not actual["message"]
+    assert actual["data"] == []
+
+
+def test_add_kairon_faq_action_with_invalid_similarity_threshold():
```
Comment (udit-pandey) [inline tests/integration_test/services_test.py:948]: add test case validating for system_prompt, context prompt with empty string

File: tests/unit_test/data_processor/data_processor_test.py [hunk 3]
```diff
@@ -117,6 +117,113 @@ async def _read_and_get_data(path: str):
 
         return _read_and_get_data
 
+    def test_add_kairon_with_invalid_similarity_threshold(self):
```
Comment (udit-pandey) [inline tests/unit_test/data_processor/data_processor_test.py:120]: add test case validating for system_prompt, context prompt with empty string. ValidationError should be thrown in both cases.
Also, add test case that you are able to add this action only when llm faq is enabled in bot settings

**Review by udit-pandey — CHANGES_REQUESTED** [review]

**Review by udit-pandey — DISMISSED** [review]

**Review by udit-pandey — APPROVED** [review]
approved
